src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManagerSession.cs

```diff
@@ -48,10 +49,26 @@ public bool RecallPublishedItemId(string id, out ContentItem item)
             return _publishedItemsById.TryGetValue(id, out item);
         }
 
+        public void AddRemovedContentItem(ContentItem item)
```

Comment:
@jtkech for infos.

This is fine, what I was suggesting, is that maybe if it was more generic, we could use it for more things (i.e. SetHomePage) - and external users could use it as well...

Something like
```
private readonly List<ContentCommand>() _commands;

public abstract class ContentCommand
{
  public ContentItem ContentItem { get; set; }
}

public class RemovedContentItemCommand : ContentItemCommand
{
}

public class SetHomepageContentItemCommand : ContentItemCommand
{
}

public bool IsCommand<T>(ContentItem contentItem)
{
  return _commands.Any(x => x.ContentItem.ContentItemId = contentItem.ContentItemId && x is typeof(T));
}
```

That kind of thing. But was just an idea, and this is fine.

Reply:
@deanmarcussen 

Okay i understand, that's a good idea, the manager session might not be the better place to cache a kind of event / command, can you suggest a change in this PR, or in a separate PR if you don't have time.

Note: We need to compare by ref on the `ContentItem` instance, not by only using `ContentItemId`, because when removing an item, multiple versions may be removed, and we want to add only one record in the index table.

A list of commands is useful for history e.g. for syncing, to deffer some works e.g. to avoid duplicate commands or in a given order, maybe other cases, otherwise an event handler is enough. So, here is it only for index providers that can't be event handlers, and e.g. `SetHomepageContentItemCommand` would not be related to an index provider.

**Update**: In fact a scoped index provider can be an `IContenHandler`, it only needs to not resolve `ISession` as it is resolved while the session itself is resolved, but not so easy as we can't inherit from the base class `ContentHandlerBase` as we already derive from the base class `IndexProvider<T>` so we have to define all the many methods. So, what i did is to use `IndexProvider` instead and then just added this

        public string CollectionName { get; set; }

        void IIndexProvider.Describe(IDescriptor context)
        {
            Describe((DescribeContext<ContentItem>)context);
        }

        public Type ForType() => typeof(ContentItem);

Then i could derive from `ContentHandlerBase` and then just override `RemovedAsync()`, just tried it works, when deleting an item this `RemovedAsync()` is executed, right now it does nothing but i will try to use it this week end.

So maybe we could revert the session manager changes, but we would have to do the same for another index provider needing it, e.g. the localization index provider.